build/Makefile

```diff
@@ -0,0 +1,167 @@
+# Adapted from https://marmelab.com/blog/2016/02/29/auto-documented-makefile.html
+help: ## Show this help message
+	@grep -E '^[a-zA-Z_-]+:.*?##.*$$' $(MAKEFILE_LIST) | awk 'BEGIN { \
+	 FS = ":.*?## "; \
+	 printf "\033[1m%-30s\033[0m %s\n", "TARGET", "DESCRIPTION" \
+	} \
+	{ printf "\033[32m%-30s\033[0m %s\n", $$1, $$2 }'
+
+.PHONY: all
+all: build dependencies artifacts checks ## Run all steps for content preparation
+
+.PHONY: build
+build: ## Build core OSCAL artifacts to convert content examples
+	$(MAKE) -C oscal/build dependencies
+	$(MAKE) -C oscal/build artifacts
+
+# Used to automatically install certain executables
+JQ_INSTALL_BIN:=jq-linux-amd64
+JQ_INSTALL_VERSION:=jq-1.7
+JQ_INSTALL_URL:=https://github.com/jqlang/jq/releases/download/$(JQ_INSTALL_VERSION)/$(JQ_INSTALL_BIN)
+JQ_INSTALL_PATH:=./jq
+JQ_PATH:=$(shell which jq || echo $(JQ_INSTALL_PATH))
+JQ_INSTALL_COMMAND=curl $(CURL_INSTALL_OPTS) -o $(JQ_INSTALL_PATH) $(JQ_INSTALL_URL) && chmod +x $(JQ_INSTALL_PATH)
+
+$(JQ_INSTALL_PATH):
+	@$(JQ_INSTALL_COMMAND)
+
+YQ_INSTALL_BIN:=yq_linux_amd64
+YQ_INSTALL_VERSION:=v4.35.2
+YQ_INSTALL_URL:=https://github.com/mikefarah/yq/releases/download/$(YQ_INSTALL_VERSION)/$(YQ_INSTALL_BIN)
+YQ_INSTALL_PATH:=./yq
+YQ_PATH:=$(shell which yq || echo $(YQ_INSTALL_PATH))
+YQ_INSTALL_COMMAND:=curl $(CURL_INSTALL_OPTS) -o $(YQ_INSTALL_PATH) $(YQ_INSTALL_URL) && chmod +x $(YQ_INSTALL_PATH)
+
+$(YQ_PATH):
+	@$(YQ_INSTALL_COMMAND)
+
+$(XMLLINT_PATH):
+	@$(XMLLINT_INSTALL_COMMAND)
+
+.PHONY: dependencies
+dependencies: $(JQ_PATH) $(XMLLINT_PATH) $(YQ_PATH) ## Install binary build jq and yq dependencies for repo
+
+.PHONY: artifacts
+artifacts: copy-readmes copy-xml-content resolve-xml-profiles convert-min-json-content reformat-json-content convert-yaml-content ## Generate all artifacts
+
+.PHONY: copy-readmes
+copy-readmes: $(GEN_READMES) ## Copy README files to release location
+
+$(GEN_CONTENT_DIR)/%.md: $(SRC_DIR)/%.md
+	# $(@D): The directory part of the file name of the target, with the
+	# trailing slash removed.
+	# https://www.gnu.org/software/make/manual/html_node/Automatic-Variables.html
+	@mkdir -p $(@D)
+	@cp $(SRC_DIR)/$*.md $(GEN_CONTENT_DIR)/$*.md
+
+.PHONY: copy-xml-content
+copy-xml-content: $(GEN_XML_FILES) ## Copy OSCAL XML files to release location
+
+$(GEN_CONTENT_DIR)/%.xml: $(SRC_DIR)/%.xml
+	@mkdir -p $(@D)
+	@cp $(SRC_DIR)/$*.xml $(GEN_CONTENT_DIR)/$*.xml
+
+.PHONY: resolve-xml-profiles
+resolve-xml-profiles: $(GEN_XML_RESOLVED_CATALOGS) ## Resolve OSCAL XML profiles for custom catalogs
+
+$(GEN_CONTENT_DIR)/%-resolved-profile_catalog.xml: $(SRC_DIR)/%_profile.xml
+	mkdir -p $(@D)
+	$(PROFILE_RESOLVER_RUNNER) $(GEN_CONTENT_DIR)/$*_profile.xml $(GEN_CONTENT_DIR)/$*-resolved-profile_catalog.xml $(PROFILE_RESOLVER_ARGS)
+	@sed -i'' -e "s|file:$(shell realpath $(GEN_CONTENT_DIR)/$*_profile.xml)|$(shell basename $*_profile.xml)|g" $(GEN_CONTENT_DIR)/$*-resolved-profile_catalog.xml	
```

Comment:
Potentially blocking: I keep getting mixed up on the target syntax, but I noticed that we are not using the copied profiles into the `GEN_CONTENT_DIR` instead of those in `SRC_DIR`. I can change this for the `$(PROFILE_RESOLVER_RUNNER)` command but not the targets above the `mkdir` command. Ensure with the team during review I am doing this correctly.

Reply:
@nikitawootten-nist I saw your review and I making changes, but to be clear: this is not an issue?